consensus/consensus_leader.go

```diff
@@ -31,13 +31,24 @@ var (
 
 // WaitForNewBlock waits for the next new block to run consensus on
 func (consensus *Consensus) WaitForNewBlock(blockChannel chan *types.Block, stopChan chan struct{}, stoppedChan chan struct{}, startChannel chan struct{}) {
+	// gensis block is the first block to be processed.
+	// But we shouldn't start consensus yet, as we need to wait for all validators
+	// received the leader's pub key which will be propogated via Pong message.
+	// After we started the first consensus, we will go back to normal case to wait
+	// for new blocks.
+	// The signal to start the first consensus right now is the sending of Pong message (SendPongMessage function in node/node_handler.go
+	// but it can be changed to other conditions later
+	first := true
 	go func() {
 		defer close(stoppedChan)
 		for {
 			select {
 			default:
-				// got the signal to start consensus
-				_ = <-startChannel
+				if first && startChannel != nil {
+					// got the signal to start consensus
+					_ = <-startChannel
+					first = false
```

Comment:
The purpose of this `first` variable is to receive from (block on) `startChannel` only on the first iteration, correct?  A more idiomatic way would be to do away with `first` boolean variable and instead nullify `startChannel`, that is:
```suggestion
					startChannel = nil
```

This way, we wouldn't enter this `if startChannel != nil { … }` block on the second and further iterations.

Reply:
different logic here.  startChannel is initialized to nil for the old networking code. Once we have libp2p fully working and tested, I'll clean up this code.